Fixes advanced options in tracking code generator

sgiehl opened the pull request:
### Description:

fixes #21063

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

sgiehl committed:
Fixes advanced options in tracking code generator

sgiehl committed:
built vue files

sgiehl committed:
hide custom vars option if they are not available

sgiehl committed:
add more tests for tracking code generator

Commit:
Update UI test screenshots

bx80 reviewed (approved):
I can confirm this allows the toggling of the cross domain linking option and also enables the 'outlinks' option at the same time. The cross domain option is correctly disabled for sites with only single URL configured. I also checked that the other option can be toggled and add their changes to the code :+1:

Perhaps it would be good to also change the English translation from 'Enables cross domain linking' to 'Enable cross domain linking' so it's more consistent with the other descriptions?

Also it might be out of scope for this ticket, but when switching sites shouldn't the advanced options be initialized with previously chosen values for the site? Currently if a user wants to update a single tracking code option then they would need to know all the other previously chosen options for the site and remember to set them all again before copying the code otherwise they would inadvertently disable things.

sgiehl committed:
updates expected UI test files

sgiehl committed:
also add some tests for cross domain linking

sgiehl committed:
change cross domain text

sgiehl committed:
updates expected UI test files

sgiehl commented:
I've changed the label text for the input and also added more UI tests to check that the cross domain option works correctly.
Will merge it once the tests passed and create a PR to backport it to 4.x-dev
